### PR TITLE
:new: :alembic: ci: Switch from TravisCI to CircleCI

### PR DESCRIPTION
**Pull request by jwflory**
This commit adds a working CI pipeline back to TeleIRC, given the recent
pricing changes to TravisCI has rendered it nonfunctional for TeleIRC.
This migrates our existing test suite to CircleCI, along with the
following changes:

* Go jobs:
	* Drop support for Go 1.14
	* Add support for Go 1.17
	* Update golangci-lint from v1.31.0 to v1.43.0
* Python job (for Sphinx documentation):
	* Update Python runtime from 3.7 to 3.10
	* Drop Pipenv for a simple requirements.txt for simplicity

### CODE REVIEW

**Comment by jwflory** [comment]
When #388 is merged, this CI pipeline will pass. See verification in this workflow, which includes the commit in PR #388:

https://app.circleci.com/pipelines/github/RITlug/teleirc/14/workflows/f5bcbec8-6031-497f-a747-6416f869308e

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      - build_docs
+
+commands:
+  golintci-lint:
+    description: Run linter checks on TeleIRC.
+    steps:
+      - checkout
+      - run: 
+          name: Download and install golintci-lint.
+          command: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sudo sh -s -- -b $(go env GOPATH)/bin v1.43.0
```
Comment (xforever1313) [inline .circleci/config.yml:24]: This is sketchy, lol.

Is there some kind of package we can install instead?

Comment (jwflory) [reply .circleci/config.yml:24]: @xforever1313 Sadly, no… this is literally upstream's recommended installation method. 😭 

https://golangci-lint.run/usage/install/#ci-installation

**Comment by jwflory** [comment]
Rebased on `main` per #388. All tests are now passing. ✅  Looks like it is working too, since @codeclimate is back to commenting on PRs again. 😀

File: .circleci/config.yml (hunk 1 above)
Comment (xforever1313) [reply .circleci/config.yml:24]: Gross, lol.

I guess one nice thing about using Circle, its not our machine that will get hacked if something sketchy happens :D.

**Review by xforever1313 — APPROVED** [review]
:+1:

**Comment by jwflory** [comment]
Woot! 🎉 I'm going to merge this one in, and rebase #372 to see if it passes with the checks we have. Then perhaps we can make a resolution there.

In the meantime, in it goes… 🎬

**Pull request merged**